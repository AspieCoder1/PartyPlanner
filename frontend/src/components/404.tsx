import * as React from 'react';
import { ReactComponent as Image } from '../img/404.svg';
import styles from './PageNotFound.module.scss';
import Header from './Header';

const PageNotFound = (): JSX.Element => {
	return (
		<>
			<Header />
			<div className={styles.noPageContainer}>
				<Image />
			</div>
		</>
	);
};

export default PageNotFound;
