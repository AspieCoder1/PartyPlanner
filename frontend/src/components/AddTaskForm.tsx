import * as React from 'react';
import { useFormik } from 'formik';
import styles from './AddTaskForm.module.scss';
import * as Yup from 'yup';
import CSS from 'csstype';
import { now } from 'lodash';

const buttonStyles: CSS.Properties = {
	color: '#ddd9da',
	border: 'none',
	background: 'none',
	fontSize: '24px',
};

const submitButton: CSS.Properties = {
	background: '#6f3473',
	border: 'none',
	color: 'white',
	fontSize: '24px',
	marginTop: '16px',
	padding: '8px',
};

type IProps = {
	closeModal: () => void;
	onSubmit: (user: AddTaskFormValues) => void;
};

interface AddTaskFormValues {
	taskname: string;
	taskdesc: string;
	taskdue: string;
}


export const AddTaskForm = (props: IProps): JSX.Element => {
	const initialValues: AddTaskFormValues = {
		taskname: '',
		taskdesc: '',
		taskdue: '',
	};

	const AddTaskSchema = Yup.object().shape({
		taskname: Yup.string().required('Required'),
		taskdesc: Yup.string().required('Required'),
	});

	const formik = useFormik({
		initialValues: initialValues,
		onSubmit: async (values: AddTaskFormValues, { setSubmitting, setStatus }) => {
			setSubmitting(true);
			const errors = await props.onSubmit(values);
			setStatus(errors);
			setSubmitting(false);
		},
		validationSchema: AddTaskSchema,

	});
	return (
		<div>
			<div className={styles.header}>
				<h1>Create Task</h1>
				<button
					style={buttonStyles}
					className={styles.closeModal}
					onClick={props.closeModal}
				>
					&times;
				</button>
			</div>

			<form className={styles.form} onSubmit={formik.handleSubmit}>
				<input
					className={styles.input}
					type='text'
					name='taskname'
					placeholder='Task Name'
					onChange={formik.handleChange}
					value={formik.values.taskname}
				/>
				<input
					className={styles.input}
					type='text'
					name='taskdesc'
					placeholder='Task Description'
					onChange={formik.handleChange}
					value={formik.values.taskdesc}
				/>
				<input
					className={styles.input}
					type='datetime-local'
					name='taskdue'
					placeholder='Due Date & Time'
					onChange={formik.handleChange}
					value={formik.values.taskdue}
				/>
				<button
					data-testid='submitButton'
					id='submit'
					style={submitButton}
					type='submit'
					disabled={formik.isSubmitting}
				>
					Add
				</button>
			</form>
		</div>
	);
};
