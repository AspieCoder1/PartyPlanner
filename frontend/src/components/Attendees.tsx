import React from 'react';
import styles from './ViewParty.module.scss';
import { uniqueId } from 'lodash';

type Props = {
	attendees: string[];
};

const Attendees = ({ attendees }: Props): JSX.Element => {
	return (
		<div className={styles.locationOrganiser}>
			<h2>Attendees</h2>
			{attendees.length > 0 ? attendees.map((attendee: string) => <p key={uniqueId()}>{attendee}</p>) : null}
		</div>
	);
};

export default Attendees;
