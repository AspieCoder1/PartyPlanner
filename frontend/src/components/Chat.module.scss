@import "../styles/colours";

.content {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  grid-column: 1 / span 6;
  height:90%;
  margin-left: 140px;
  margin-right: 140px;
  flex-grow: 1;
  margin-bottom: 32px;
}

.chatArea {
  padding: 16px 8px 8px 8px;
  row-gap: 8px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  background: lighten($grey, 10);
  flex-grow: 1;
  overflow: auto;
}

.chatSubmit {
  padding: 8px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 16px;
  border: $grey 1px solid;
  background: white;
}

.noMessages {
  justify-items: center;
  justify-content: center;
}

.chatInput {
  font-size: 16px;
  padding: 8px;
  border: none;
  border-bottom: $grey 1px solid;
  outline: none;
  color: $black;
  flex-grow: 1;
  margin-right: 64px;
}

.chatButton {
  margin: 0px;
  border: none;
  background: $dark-purple;
  color: white;
  font-weight: 500;
  padding-left: 16px;
  padding-right: 16px;
}

.chatInput::placeholder {
  color: $grey;
  font-size: 16px;
}

.message {
  margin-top: 0px;
  margin-bottom: 0px;
  background: $dark-purple;
  border-radius: 8px;
  padding: 8px 16px 8px 16px;
  margin-left: auto;
  max-width: 500px;
}


.otherUserMessage {
  background: white;
  border-radius: 8px;
  padding: 8px 16px 8px 16px;
  margin-right: auto;
  max-width: 500px;
}

.messageOtherText {
  margin-top: 0px;
  margin-bottom: 0px;
  color: $black;
  font-weight: 300;
  font-size: 16px;
}

.messageText {
  margin-top: 0px;
  margin-bottom: 0px;
  color: white;
  font-weight: 300;
  font-size: 16px;
}

.messageName {
  margin-top: 0px;
  margin-bottom: 0px;
  color: white;
  font-weight: 400;
  font-size: 16px;
}

.messageOtherName {
  margin-top: 0px;
  margin-bottom: 0px;
  color: $black;
  font-weight: 400;
  font-size: 16px;
}