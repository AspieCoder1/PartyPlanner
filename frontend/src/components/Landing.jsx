"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Landing_module_scss_1 = require("./Landing.module.scss");
var LandingHeader_1 = require("./LandingHeader");
var landingImage_svg_1 = require("../img/landingImage.svg");
var Landing = /** @class */ (function (_super) {
    __extends(Landing, _super);
    function Landing() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Landing.prototype.render = function () {
        return (<div>
        <LandingHeader_1.default />
        <div className={Landing_module_scss_1.default.container}>
          <h2 className={Landing_module_scss_1.default.title}>
            Take the stress out of party planning
          </h2>
          <p className={Landing_module_scss_1.default.paragraph}>
            We all know party planning is hard. But it does not have to be.
            PartyPlanner.io provides you with all the tools you need in order to
            have a stress-free party. So relax, get a drink and give
            PartyPlanner a try!
          </p>
          <button className={Landing_module_scss_1.default.button}>Get PartyPlanner</button>
          <landingImage_svg_1.default className={Landing_module_scss_1.default.img} src={landingImage_svg_1.default} alt=""/>
        </div>
      </div>);
    };
    return Landing;
}(React.Component));
exports.default = Landing;
