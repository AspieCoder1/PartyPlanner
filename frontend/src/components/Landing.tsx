import * as React from "react";
import styles from "./Landing.module.scss";
import LandingHeader from "./LandingHeader";
import img from "../img/landingImage.svg" ;
import ReactModal from 'react-modal';
import { LoginForm } from "./LoginForm";

type IState = {
  loginModalOpen: boolean;
}

export default class Landing extends React.Component<any, IState> {

  onLoginModelClose = () => {
    this.setState({loginModalOpen: false})
  }

  openLoginModel = () => {
    this.setState({loginModalOpen: true})
  }

  state: IState = {
    loginModalOpen: false
  }

  render() {
    return (
      <div>
        <LandingHeader onClickLogin={this.openLoginModel} />
        <div className={styles.container}>
          <h2 className={styles.title}>
            Take the stress out of party planning
          </h2>
          <p className={styles.paragraph}>
            We all know party planning is hard. But it does not have to be.
            PartyPlanner.io provides you with all the tools you need in order to
            have a stress-free party. So relax, get a drink and give
            PartyPlanner a try!
          </p>
          <button className={styles.button}>Get PartyPlanner</button>
          <img className={styles.img} src={img} alt="" />
        </div>
        {/*TODO: add styling to ReactModal to make it smaller and fit better*/}
        <ReactModal overlayClassName={styles.overlay} className={styles.modal} isOpen={this.state.loginModalOpen}>
          <LoginForm  closeModal={this.onLoginModelClose}/>
        </ReactModal>
      </div>
    );
  }
}
