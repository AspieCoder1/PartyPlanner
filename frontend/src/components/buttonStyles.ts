import CSS from 'csstype';

export const closeButton: CSS.Properties = {
	color: '#ddd9da',
	border: 'none',
	background: 'none',
	fontSize: '24px',
};

export const submitButton: CSS.Properties = {
	background: '#6f3473',
	border: 'none',
	color: 'white',
	fontSize: '24px',
	marginTop: '16px',
	padding: '8px',
};
