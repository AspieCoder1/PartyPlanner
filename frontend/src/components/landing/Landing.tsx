import React, { useEffect, useState } from 'react';
import styles from './Landing.module.scss';
import LandingHeader from './LandingHeader';
import img from '../../img/landingImage.svg';
import avatar from '../../img/avatar.svg';
import luke from '../../img/luke.svg';
import jeremy from '../../img/jeremy.svg';
import abbas from '../../img/abbas.svg';
import divin from '../../img/divin.svg';
import andrew from '../../img/andrew.svg';
import ReactModal from 'react-modal';
import { LoginForm } from './LoginForm';
import { RegisterForm } from './RegisterForm';
import { useSelector } from 'react-redux';
import { Store } from '../../redux/store';
import history from '../../utils/history';

const Landing = (): JSX.Element => {
	// State
	const status = useSelector((state: Store) => state.user.status);
	const [loginModalOpen, setLoginModalOpen] = useState<boolean>(false);
	const [registerModalOpen, setRegisterModalOpen] = useState<boolean>(false);

	// Modal controls
	const openLoginModal = (): void => {
		setLoginModalOpen(true);
	};
	const closeLoginModal = (): void => {
		setLoginModalOpen(false);
	};
	const openRegisterModal = (): void => {
		setRegisterModalOpen(true);
	};
	const closeRegisterModal = (): void => {
		setRegisterModalOpen(false);
	};

	useEffect(() => {
		if (status === 'success') {
			history.push('/dashboard');
		}
	}, [status]);

	return (
		<>
			<LandingHeader onClickLogin={openLoginModal} onClickRegister={openRegisterModal} />
			<div className={styles.container}>
				<h2 className={styles.title}>Take the stress out of party planning</h2>
				<p className={styles.paragraph}>
					We all know party planning is hard. But it does not have to be. PartyPlanner.io provides you with all the
					tools you need in order to have a stress-free party. So relax, get a drink and give PartyPlanner a try!
				</p>
				<button className={styles.button} onClick={openRegisterModal}>
					Get PartyPlanner
				</button>
				<img className={styles.img} src={img} alt='' />
			</div>
			<br/>
			<div className={styles.about}>
				<h1 className={styles.aboutUs}>About Us</h1>
				<div className={styles.person}>
					<h2>Luke Braithwaite</h2>
					<img className={styles.avatar} src={luke} alt='avatar' />
					<p>I am Luke from Manchester, UK.
						<br/>I have experience with Server-side development and has had DevOps experience</p>
				</div>
				<div className={styles.person}>
					<h2>Jeremy Roe</h2>
					<img className={styles.avatar} src={jeremy} alt='avatar' />
					<p>I am Jeremy from London, UK.
						<br/>I have experience with Python and SQL.
						<br/>In my spare time I learn Japanese and French.
					</p>
				</div>
				<div className={styles.person}>
					<h2>Divin Jacob</h2>
					<img className={styles.avatar} src={divin} alt='avatar' />
					<p>Hi, my name is Divin, I am from Manchester.
						<br/>In my spare time I enjoy playing badminton and watching football.
						<br/>I have experience with .NET</p>
				</div>
				<div className={styles.person}>
					<h2>Abbas Sayed</h2>
					<img className={styles.avatar} src={abbas} alt='avatar' />
					<p>My name is Abbas and I am from Afghanistan!
						<br/>I am experienced in Web and Android/iOS Development.
						<br/>Unlike Luke, I love PHP so much &lt;3</p>
				</div>
				<div className={styles.person}>
					<h2>Radu Pirlog</h2>
					<img className={styles.avatar} src={avatar} alt='avatar' />
					<p>Hello! I am Radu from Romania!
						<br/>I have experience with C++ and Media editing.
						<br/>In my free time, I like to go for a drive and relax. </p>
				</div>
				<div className={styles.person}>
					<h2>Andrew</h2>
					<img className={styles.avatar} src={andrew} alt='avatar' />
					<p>Insert info about Andrew here</p>
				</div>
			</div>
			<ReactModal
				overlayClassName={styles.overlay}
				className={styles.modal}
				isOpen={loginModalOpen}
				ariaHideApp={false}
			>
				<LoginForm closeModal={closeLoginModal} />
			</ReactModal>
			<ReactModal
				overlayClassName={styles.overlay}
				className={styles.modal}
				isOpen={registerModalOpen}
				ariaHideApp={false}
			>
				<RegisterForm closeModal={closeRegisterModal} />
			</ReactModal>
		</>
	);
};
export default Landing;
