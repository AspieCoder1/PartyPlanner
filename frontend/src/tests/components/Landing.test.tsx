import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Landing } from '../../components/Landing';

describe('Landing page component', () => {
	it('Should render component correctly', () => {
		const tree = renderer.create(<Landing registerUser={jest.fn()} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
