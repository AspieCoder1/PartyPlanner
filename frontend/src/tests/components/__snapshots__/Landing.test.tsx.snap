// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component Should render component correctly 1`] = `
<div>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        PartyPlanner.io
      </h1>
      <div
        className="headerButtons"
      >
        <button
          className="loginButton"
          onClick={[Function]}
        >
          Login
        </button>
        <button
          className="registerButton"
          onClick={[Function]}
        >
          Register
        </button>
      </div>
    </div>
  </header>
  <div
    className="container"
  >
    <h2
      className="title"
    >
      Take the stress out of party planning
    </h2>
    <p
      className="paragraph"
    >
      We all know party planning is hard. But it does not have to be. PartyPlanner.io provides you with all the tools you need in order to have a stress-free party. So relax, get a drink and give PartyPlanner a try!
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Get PartyPlanner
    </button>
    <img
      alt=""
      className="img"
      src="landingImage.svg"
    />
  </div>
</div>
`;
